File: app/src/main/java/com/example/data/model/WillRyanProfileData.kt
package com.example.data.model

import androidx.compose.runtime.mutableStateListOf

object WillRyanProfileData {
    val profileName = "Local Profile"
    
    // Subscriptions are loaded dynamically and stored in private local database
    val subscribedChannels = mutableStateListOf<String>()

    fun isSubscribed(channelName: String): Boolean {
        if (channelName.isBlank()) return false
        val trimmed = channelName.trim()
        return subscribedChannels.any { it.equals(trimmed, ignoreCase = true) }
    }

    fun addSubscribedChannel(name: String) {
        val trimmed = name.trim()
        if (trimmed.isNotBlank() && !isSubscribed(trimmed)) {
            subscribedChannels.add(0, trimmed)
        }
    }

    fun removeSubscribedChannel(name: String) {
        val trimmed = name.trim()
        subscribedChannels.removeAll { it.equals(trimmed, ignoreCase = true) }
    }

    fun clearAllSubscribedChannels() {
        subscribedChannels.clear()
    }
}

File: windows/VixzDesktop/Models/WillRyanProfileData.cs
using System;
using System.Collections.Generic;
using System.Linq;
using VixzDesktop.Services;

namespace VixzDesktop.Models
{
    public static class WillRyanProfileData
    {
        public static string ProfileName => "Local Profile";

        public static readonly List<string> DefaultSubscribedChannels = new List<string>();

        public static List<string> SubscribedChannels
        {
            get
            {
                if (StorageService.Settings.SubscribedChannels == null)
                {
                    StorageService.Settings.SubscribedChannels = new List<string>();
                }
                return StorageService.Settings.SubscribedChannels;
            }
        }

        public static bool IsSubscribed(string channelName)
        {
            if (string.IsNullOrWhiteSpace(channelName)) return false;
            var trimmed = channelName.Trim();
            return SubscribedChannels.Any(c => c.Equals(trimmed, StringComparison.OrdinalIgnoreCase));
        }

        public static void AddSubscribedChannel(string name)
        {
            var trimmed = name.Trim();
            if (!string.IsNullOrWhiteSpace(trimmed) && !IsSubscribed(trimmed))
            {
                SubscribedChannels.Insert(0, trimmed);
                StorageService.Save();
            }
        }

        public static void RemoveSubscribedChannel(string name)
        {
            var trimmed = name.Trim();
            SubscribedChannels.RemoveAll(c => c.Equals(trimmed, StringComparison.OrdinalIgnoreCase));
            StorageService.Save();
        }

        public static void ClearAllSubscribedChannels()
        {
            SubscribedChannels.Clear();
            StorageService.Save();
        }

        public static void RestoreDefaultChannels()
        {
            SubscribedChannels.Clear();
            StorageService.Save();
        }
    }
}
